cachix/src/Cachix/Client/PushQueue.hs

```diff
@@ -112,6 +116,8 @@ exitOnceQueueIsEmpty stopProducerCallback pushWorker queryWorker queryWorkerStat
           cancelWith queryWorker StopWorker
           cancelWith pushWorker StopWorker
         else do
+          -- extend shutdown for another 90s
+          Systemd.notify False $ "EXTEND_TIMEOUT_USEC=" <> show (90 * 1000 * 1000)
```

Comment:
I was not sure how systemd would interpret this timeout but what it does is according to the source:
`current_time + EXTEND_TIMEOUT_USEC`. So this implementation seems correct: if there is no cachix update for 90s
than the service is killed.

Reply:
That was my interpretation from `the service manager will allow the service to
           continue to stop, provided the service repeats
           "EXTEND_TIMEOUT_USEC=..."  within the interval specified, or
           terminates itself (see sd_notify(3)).`

Reply:
@Mic92 if this looks good to you I'll merge it and you can give it a try :)

Reply:
It does.